### DISCUSSION TITLE
[@LekoArts/gatsby-theme-cara]: from ./node_modules/gatsby/dist/utils/babel-loader => projects.mdx: Unexpected token #36579

### DISCUSSION THREAD

**Discussion by eniger31** [LekoArts/gatsby-themes#985] [Questions]
## Description

Hi, I have a problem on @LekoArts/gatsby-theme-cara. 

 **```
ERROR 

Module build failed (from ./node_modules/gatsby/dist/utils/babel-loader.js):
SyntaxError: /Users/Regine/GATSBY/MonSiteRG31/src/@lekoarts/gatsby-theme-cara/sections/about.mdx: Unexpected token (1:0)

> 1 | ## About
```**

Any ideas? 😅
Thanks

### Steps to reproduce

`Gatsby develop `
`https://github.com/RG31/MonSiteRG31`

### Expected result

What should happen? GAtsby deveop RUN correctly

### Actual result

 **```
ERROR 

Module build failed (from ./node_modules/gatsby/dist/utils/babel-loader.js):
SyntaxError: /Users/Regine/GATSBY/MonSiteRG31/src/@lekoarts/gatsby-theme-cara/sections/about.mdx: Unexpected token (1:0)

> 1 | ## About
```**

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here.
`Regine@macbook-pro-regine MonSiteRG31 % gatsby info --clipboard

  System:
    OS: macOS 10.15.7
    CPU: (8) x64 Intel(R) Core(TM) i5-8257U CPU @ 1.40GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 14.15.5 - /usr/local/bin/node
    npm: 8.5.5 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 105.0.5195.102
    Safari: 15.6.1
  npmPackages:
    gatsby: ^4.18.2 => 4.22.1
    gatsby-plugin-flow: ^3.22.0 => 3.22.0
    gatsby-plugin-gatsby-cloud: ^4.18.1 => 4.22.0
    gatsby-plugin-manifest: ^4.18.1 => 4.22.0
    gatsby-plugin-mdx: ^4.1.1 => 4.1.1
    gatsby-plugin-webpack-bundle-analyser-v2: ^1.1.27 => 1.1.27
    gatsby-source-filesystem: ^4.22.0 => 4.22.0
  npmGlobalPackages:
    gatsby-cli: 4.22.1`

**Comment by LekoArts** [comment]
You have to use v3 of `gatsby-plugin-mdx` (that's also what the theme installs). It's not ready for v4 yet (progress at https://github.com/LekoArts/gatsby-themes/pull/967)

**Reply by eniger31** [reply]
Many thanks 🖤 @LekoArts

**Reply by eniger31** [reply]
How can I downgrade Gatsby to a specific version ?

**Reply by eniger31** [reply]
1. Open your package.json file and change the actual version you want to use.
2. Then run npm update to make sure you update all the packages to the right version.

**Reply by eniger31** [reply]
Sorry but I tried several installation options without success . `npm install gatsby-plugin-mdx@v3.0.0` `npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: cara@0.2.11
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"^18.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.13.1 || ^17.0.0" from @mdx-js/react@1.6.22
npm ERR! node_modules/@mdx-js/react
npm ERR!   peer @mdx-js/react@"^1.0.0" from gatsby-plugin-mdx@3.0.0
npm ERR!   node_modules/gatsby-plugin-mdx
npm ERR!     gatsby-plugin-mdx@"v3.0.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/Regine/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/Regine/.npm/_logs/2022-09-09T13_59_29_826Z-debug-0.log
Regine@macbook-pro-regine gatsby-starter-portfolio-cara % npm install gatsby-plugin-mdx@v3.1.1
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: cara@0.2.11
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"^18.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.13.1 || ^17.0.0" from @mdx-js/react@1.6.22
npm ERR! node_modules/@mdx-js/react
npm ERR!   peer @mdx-js/react@"^1.0.0" from gatsby-plugin-mdx@3.1.1
npm ERR!   node_modules/gatsby-plugin-mdx
npm ERR!     gatsby-plugin-mdx@"v3.1.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/Regine/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/Regine/.npm/_logs/2022-09-09T13_59_43_791Z-debug-0.log
Regine@macbook-pro-regine gatsby-starter-portfolio-cara % npm install gatsby-plugin-mdx @mdx-js/mdx@v3 @mdx-js/react@v3
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: cara@0.2.11
npm ERR! Found: @mdx-js/react@undefined
npm ERR! node_modules/@mdx-js/react
npm ERR!   @mdx-js/react@"v3" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @mdx-js/react@"^2.0.0" from gatsby-plugin-mdx@4.1.1
npm ERR! node_modules/gatsby-plugin-mdx
npm ERR!   gatsby-plugin-mdx@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/Regine/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/Regine/.npm/_logs/2022-09-09T14_01_53_469Z-debug-0.log
`

**Reply by LekoArts** [reply]
As the error suggests, use `npm install gatsby-plugin-mdx@^3 --legacy-peer-deps`

**Comment by eniger31** [comment]
Ok for the solution Many thanks 🖤 . BUT when "gatsby develop" run `success run page queries - 0.020s - 3/3 150.36/s

 ERROR 

/Users/Regine/GATSBY/gatsby-starter-portfolio-cara/src/@lekoarts/gatsby-theme-cara/sections/projects.mdx: Identifier '_frontmatter' has already been declared. (2:13)

  1 | export const _frontmatter = {};
> 2 | export const _frontmatter = {}
    |              ^



  SyntaxError: /Users/Regine/GATSBY/gatsby-starter-portfolio-cara/src/@lekoarts/gatsby-theme-cara/sections/projects.mdx: Identifier '_frontmatter' has already been declared. (2:13)
    1 | export const _frontmatter = {};
  > 2 | export const _frontmatter = {}
      |              ^
  
  - credentials.ts:62 instantiate
    [gatsby-starter-portfolio-cara]/[@babel]/parser/src/parse-error/credentials.ts:62:21`   ** there is no variable in .mdx files** 🤫

**Reply by eniger31** [reply]
@LekoArts I do not have access to the sources to modify this problem. can you please help me?  `ERROR 

/Users/Regine/GATSBY/gatsby-starter-portfolio-cara/src/@lekoarts/gatsby-theme-cara/sections/projects.mdx: Identifier '_frontmatter' has already been declared. (2:13)

  1 | export const _frontmatter = {};
> 2 | export const _frontmatter = {}
    |              ^



  SyntaxError: /Users/Regine/GATSBY/gatsby-starter-portfolio-cara/src/@lekoarts/gatsby-theme-cara/sections/projects.mdx: Identifier '_frontmatter' has already been declared. (2:13)
    1 | export const _frontmatter = {};
  > 2 | export const _frontmatter = {}
      |              ^`

**Comment by LekoArts** [comment]
Please share your repository, otherwise I can't help

**Reply by eniger31** [reply]
🖤 Thank you @LekoArts . this is my repository https://github.com/RG31/gatsby-starter-portfolio-cara.git

**Reply by eniger31** [reply]
so  I  I added the option mdx=false and it works `  plugins: [
    {
      resolve: `@lekoarts/gatsby-theme-cara`,
      // See the theme's README for all available options
      options: {
        mdx: false
      },
    },`

**Reply by eniger31** [reply]
[SyntaxError: unknown: Identifier '_frontmatter' has already been declared](https://github.com/ChristopherBiscardi/gatsby-mdx/issues/354#top)
#354

**Comment by eniger31** [comment]
So thank you @LekoArts , you can close this issue 
🖤 Thanks
